src/libgrate/grate-3d.c

```diff
@@ -145,6 +145,25 @@ static void grate_3d_set_guardband(struct host1x_pushbuf *pb,
 	}
 }
 
+static int get_cull_face(struct grate_3d_ctx *ctx)
+{
+	switch (ctx->cull_face) {
+	case GRATE_CULL_FACE_NONE:
+		return CULL_FACE_NONE;
+
+	case GRATE_CULL_FACE_FRONT:
+		return ctx->tri_face_front_cw ? CULL_FACE_CW : CULL_FACE_CCW;
+
+	case GRATE_CULL_FACE_BACK:
+		return ctx->tri_face_front_cw ? CULL_FACE_CCW : CULL_FACE_CW;
+
+	case GRATE_CULL_FACE_BOTH:
+		return CULL_FACE_BOTH;
+	}
+
+	return CULL_FACE_NONE; /* just a stupid fallback */
```

Comment:
And yeah, I guess it would make sense to print (or just assert) in default instead.